src/main/java/com/unzer/payment/communication/JsonURLConverter.java

```diff
@@ -45,7 +45,7 @@ public URL deserialize(JsonElement json, Type typeOfT, JsonDeserializationContex
 		try {
 			return new URL(urlValue);
 		} catch (MalformedURLException e) {
-			logger.warn("Invalid URL '" + urlValue + "': " + e.getMessage());
+			logger.warn("Invalid URL '{}': {}", urlValue, e.getMessage());
```

Comment:
Doesnt it cause any problem if no logger is provided?
Doesnt the sdk still depend on the log4j? Meaning if its not part of the integrator projekt this call would cause an exception, right?